src/AndersonAcceleration.f90

```diff
@@ -205,40 +219,36 @@ SUBROUTINE step_AndersonAccelerationType(solver,x_new)
     depth_s=MIN(solver%depth,solver%s-solver%start)
     !Push back vectors to make room for new ones
     DO i=depth_s,1,-1
-      solver%Gx(:,i+1)=solver%Gx(:,i)
-      solver%r(:,i+1)=solver%r(:,i)
+      CALL BLAS_copy(solver%Gx(i),solver%Gx(i+1))
+      CALL BLAS_copy(solver%r(i),solver%r(i+1))
     ENDDO
 
     !Set new vectors based on input
-    solver%Gx(:,1)=x_new(:)
-    solver%r(:,1)=solver%Gx(:,1)-solver%x(:,1)
+    CALL BLAS_copy(x_new,solver%Gx(1))
+    CALL BLAS_copy(solver%Gx(1),solver%r(1))
+    CALL BLAS_axpy(solver%x(1),solver%r(1),-1.0_SRK)
 
     !Get fit coefficients
     IF(depth_s == 1) THEN
       !Depth 1 is an especially simple case, where alpha can be calculated with a simple formula
-      tmpA=0.0_SRK
-      tmpb=0.0_SRK
-      DO j=1,solver%n
-        tmp=solver%r(j,2)-solver%r(j,1)
-        tmpA=tmpA+tmp*tmp
-        tmpb=tmpb+solver%r(j,2)*tmp
-      ENDDO
+      CALL BLAS_copy(solver%r(2),tmpvec)
+      CALL BLAS_axpy(solver%r(1),tmpvec,-1.0_SRK)
+      tmpA=BLAS_dot(tmpvec,tmpvec)
+      tmpb=BLAS_dot(solver%r(2),tmpvec)
       solver%alpha(1)=tmpb/tmpA
     ELSEIF(depth_s > 1) THEN
       !Construct coefficient matrix, right hand side, and solve for fit coefficients
       DO i=1,depth_s
+        CALL BLAS_copy(solver%r(i),x_new)
```

Comment:
You made a comment before that the blas routines make the math less readable... could always add a comment to communicate exactly what math you are trying to implement

Reply:
Since the math is simple enough here I don't think we need to go that far.